core/src/test/java/org/apache/iceberg/TestMergeAppend.java

```diff
@@ -92,6 +94,74 @@ public void testEmptyTableAppend() {
         statuses(Status.ADDED, Status.ADDED));
   }
 
+  @Test
+  public void testEmptyTableAppendFilesWithDifferentSpecs() {
```

Comment:
Can we add a test where validations will fail when there are multiple specs? I know we're exercising that for a single spec now but it'd be nice to see that we get the expected failures when there's multiple.